src/game/combat.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright � 2020 uwadmin12
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <unordered_set>
+#include <map>
+
+#include "types.h"
+#include "object/creature.h"
+
+namespace reone {
+
+namespace game {
+
+class Area;
+
+class Combat {
+
+public:
+    Combat(Area *area);
+
+    void update(float dt);
+
+    /*
+    * Roles:
+    * 1. Scan surroundings for hostiles
+    * 2. Set attack target
+    * 3. Queue Commands (e.g. go to, item consumption, equipment swapping etc.)
+    */
+    void queueInstructions(std::shared_ptr<Creature> &combatant);
+
+    /*
+    * Roles:
+    * 1. Timed Animation Interrupt Control
+    * 2. Immediate signal exchange between combatants
+    */
+    void combatStateMachine(std::shared_ptr<Creature> &combatant);
+
+    /*
+    * Roles:
+    * 1. Animate dueling and isolated attacks
+    * 2. Animate stagger
+    * 
+    * [Future Roles]:
+    * 1. Evaluate the possibility and extent of damage infliction
+    * 2. Evaluate the possibility and duration of stagger
+    * 3. Share the responsibility of setting timers
+    */
+    void animationControl();
+
+private:
+    Area*                                                       _area;
```

Comment:
nit: Let's not pad variable names. I find it time consuming and not really beneficial to readability.